src/memory.hpp

```diff
@@ -428,13 +428,18 @@ struct cvk_sampler : public _cl_sampler, api_object<object_magic::sampler> {
                 std::vector<cl_sampler_properties>&& properties)
         : api_object(context), m_normalized_coords(normalized_coords),
           m_addressing_mode(addressing_mode), m_filter_mode(filter_mode),
-          m_properties(std::move(properties)), m_sampler(VK_NULL_HANDLE) {}
+          m_properties(std::move(properties)), m_sampler(VK_NULL_HANDLE),
+          m_sampler_norm(VK_NULL_HANDLE) {}
 
     ~cvk_sampler() {
         if (m_sampler != VK_NULL_HANDLE) {
             auto vkdev = context()->device()->vulkan_device();
             vkDestroySampler(vkdev, m_sampler, nullptr);
         }
+        if (m_sampler_norm != VK_NULL_HANDLE) {
+            auto vkdev = context()->device()->vulkan_device();
```

Comment:
Maybe that line could be common and outside the if's?

Reply:
Done